Lib/inspect.py

```diff
@@ -834,9 +834,8 @@ def _finddoc(obj):
             cls = self.__class__
     # Should be tested before isdatadescriptor().
     elif isinstance(obj, property):
-        func = obj.fget
-        name = func.__name__
-        cls = _findclass(func)
+        name = obj.__name__
```

Comment:
Note that it can now be None.

Reply:
Good point. Shall I add a fallback to `func.__name__`? I can also add this fallback into the getter for `__name__`, but that may be too much?

Reply:
Another option is to initialize `__name__` to `fget.__name__` in the constructor.

Reply:
And since `fget` may not have the `__name__` attribute, we need to decide what to do in this case. Yest another option -- leave it uninitialized. I do not know what is better.

What other `__name__` attribute can be None or raise AttributeError?

Reply:
There is a subtle difference with `singledispatch`, that the last name takes priority. With `singledispatch` you can write:
```py
@singledispatch
def foo(a): ...

@foo.register(int)
def bar(a): ...
```
and then use `foo()`. It is common to use `_` as a placeholder for additional dispatches.

But if you write
```py
@property
def foo(self): ...

@foo.setter
def bar(self, value): ...
```
you can only use `bar` as writeable property. `foo` is a readonly property. Of course nobody uses it, all repeat the same name for getter, setter and deleter.

Should we support such weird case? I do not think so. But the current code which prefers the name of the getter is less founded. It only works if the same name is used for getter, setter, and deleter. So it does not really matter. We have no good use case to support.

 We can add `__set_name__` to classmethod and staticmethod if there is a benefit.

BTW, I consider to add `__objclass__` and `__qualname__` (as dynamic attribute based on `__objclass__` and `__name__`) to property. This can make property more compatible with builtin attributes and can allow to generalize some code. It may be not enough, we may need a general way to distinguish read-only attributes from writeable attributes.